### PR TITLE
Consistently use `bytes` not `byte_slice`

### PR DESCRIPTION
**Pull request by Manishearth**
Fixes https://github.com/unicode-org/icu4x/issues/5815

### CODE REVIEW

**Comment by Manishearth** [comment]
Reviewer note: can be reviewed individually. Do take a look at _non-zerovec_ code affected by each commit in case something incorrect snuck through.

File: components/casemap/src/provider/data.rs [hunk 1]
```diff
@@ -367,7 +367,7 @@ impl CaseMapDataULE {
 /// 6. The equality invariant is satisfied
 unsafe impl ULE for CaseMapDataULE {
     fn validate_byte_slice(bytes: &[u8]) -> Result<(), UleError> {
-        let sixteens = RawBytesULE::<2>::parse_byte_slice(bytes)?;
+        let sixteens = RawBytesULE::<2>::parse_bytes(bytes)?;
```
Comment (sffc) [inline components/casemap/src/provider/data.rs:370]: Thought: could be `try_from_bytes`

File: utils/zerovec/src/cow.rs [hunk 2]
```diff
@@ -97,7 +97,7 @@ impl<'a, V: VarULE + ?Sized> VarZeroCow<'a, V> {
     }
 
     /// Construct from an owned slice. Errors if the slice doesn't represent a valid `V`
-    pub fn parse_owned_byte_slice(bytes: Box<[u8]>) -> Result<Self, UleError> {
+    pub fn parse_owned_bytes(bytes: Box<[u8]>) -> Result<Self, UleError> {
```
Comment (sffc) [inline utils/zerovec/src/cow.rs:100]: Suggestion (optional): `try_from_boxed_bytes` or `parse_boxed_bytes`. I think "boxed bytes" is language used elsewhere

File: components/experimental/src/dimension/provider/extended_currency.rs [hunk 3]
```diff
@@ -71,10 +69,10 @@ impl databake::Bake for CurrencyExtendedDataV1<'_> {
     fn bake(&self, ctx: &databake::CrateEnv) -> databake::TokenStream {
         use zerovec::ule::VarULE;
         ctx.insert("icu_experimental::dimension::provider::extended_currency");
-        let bytes = self.display_names.elements.as_byte_slice().bake(ctx);
-        // Safety: The bytes are returned by `PluralElementsPackedULE::as_byte_slice`.
+        let bytes = self.display_names.elements.as_bytes().bake(ctx);
+        // Safety: The bytes are returned by `PluralElementsPackedULE::slice_as_bytes`.
```
Comment (sffc) [inline components/experimental/src/dimension/provider/extended_currency.rs:73]:
```suggestion
        // Safety: The bytes are returned by `PluralElementsPackedULE::as_bytes`.
```

File: utils/zerovec/src/ule/mod.rs [hunk 4]
```diff
@@ -87,26 +87,26 @@ where
...
-    /// The default implementation executes [`Self::validate_byte_slice()`] followed by
-    /// [`Self::from_byte_slice_unchecked`].
+    /// The default implementation executes [`Self::validate_bytes()`] followed by
+    /// [`Self::from_bytes_unchecked`].
     ///
     /// Note: The following equality should hold: `bytes.len() % size_of::<Self>() == 0`. This
     /// means that the returned slice can span the entire byte slice.
-    fn parse_byte_slice(bytes: &[u8]) -> Result<&[Self], UleError> {
-        Self::validate_byte_slice(bytes)?;
+    fn parse_bytes(bytes: &[u8]) -> Result<&[Self], UleError> {
```
Comment (sffc) [inline utils/zerovec/src/ule/mod.rs:102]: Suggestion: `parse_bytes_to_slice`?

Or `try_slice_from_bytes` which is a nicer parallel to `slice_as_bytes`

File: utils/zerovec/src/ule/mod.rs [hunk 5]
```diff
@@ -127,10 +127,10 @@ where
     ///
     /// Safety checklist:
     ///
-    /// 1. This method *must* return the same result as [`Self::parse_byte_slice()`].
+    /// 1. This method *must* return the same result as [`Self::parse_bytes()`].
     /// 2. This method *must* return a slice to the same region of memory as the argument.
     #[inline]
-    unsafe fn from_byte_slice_unchecked(bytes: &[u8]) -> &[Self] {
+    unsafe fn from_bytes_unchecked(bytes: &[u8]) -> &[Self] {
```
Comment (sffc) [inline utils/zerovec/src/ule/mod.rs:133]: Suggestion: `slice_from_bytes_unchecked`?

**Review by sffc — DISMISSED** [review]

File: utils/zerovec/src/cow.rs (hunk 2 above)
Comment (Manishearth) [reply utils/zerovec/src/cow.rs:100]: I think for a Cow type "owned" is probably better.

File: components/casemap/src/provider/data.rs (hunk 1 above)
Comment (Manishearth) [reply components/casemap/src/provider/data.rs]: I like this being `parse()` but it's worth discussing. I'd want to do that separately IMO.

**Review by sffc — APPROVED** [review]

**Pull request merged**